modules/firefox/hm.nix

```
@@ -1,36 +1,66 @@
-# Consider also updating the LibreWolf module when updating this module,
-# as they are very similar.
-
-{ config, lib, ... }:
+{ config, lib, ...}:
 
 let
-  profileSettings = {
-    settings = {
-      "font.name.monospace.x-western" = config.stylix.fonts.monospace.name;
-      "font.name.sans-serif.x-western" = config.stylix.fonts.sansSerif.name;
-      "font.name.serif.x-western" = config.stylix.fonts.serif.name;
-    };
-  };
-  makeProfileSettingsPair = profileName:
-    lib.nameValuePair profileName profileSettings;
-  derivatives = [
-    {path = "firefox"; name = "Firefox";}
-    {path = "librewolf"; name = "LibreWolf";}
+  targets = [
+    { path = "firefox"; name = "Firefox"; }
+    { path = "librewolf"; name = "LibreWolf"; }
   ];
+  eachConfig = mkCfg: targets: lib.mkMerge (map mkCfg targets);
+  eachTarget = mkCfg: lib.mkIf config.stylix.enable (eachConfig (target: let
+    cfg = config.stylix.targets.${target.path};
+    programCfg = config.programs.${target.path};
+  in
+    lib.mkIf cfg.enable (mkCfg {inherit target cfg programCfg;})
+  ) targets);
 in {
-  options.stylix.targets = lib.listToAttrs (map (drv: lib.nameValuePair drv.path {
-    enable =
-      config.lib.stylix.mkEnableTarget drv.name true;
+  options.stylix.targets = lib.listToAttrs (map (target:
+    lib.nameValuePair target.path {
+      enable = config.lib.stylix.mkEnableTarget target.name true;
+
+      profileNames = lib.mkOption {
+        description = "The ${target.name} profile names to apply styling on.";
+        type = lib.types.listOf lib.types.str;
+        default = [];
+      };
+
+      firefoxGnomeTheme.enable = config.lib.stylix.mkEnableTarget "Firefox GNOME theme" false;
```

IIUC, the [firefox-gnome-theme](https://github.com/rafaelmardojai/firefox-gnome-theme) theme is applied to all valid Firefox derivatives. As long as the "valid Firefox derivative" predicate is uphold by Home Manager and Stylix, the specific derivative instance does not matter and we can simplify the documentation to:

```suggestion
      gnomeTheme.enable = config.lib.stylix.mkEnableTarget "GNOME theme" false;
```

---

"Firefox GNOME theme" is the name that project had given itself. If we want to use a more descriptive name I would even suggest calling it "GTK theme". I get that the name is confusing, but IMHO I'd rather keep using the official name of that project.

---

Agreed. To avoid any confusion, we could simply link to the project:

```suggestion
      firefoxGnomeTheme.enable =
        config.lib.stylix.mkEnableTarget
        ''
          [Firefox GNOME
          theme](https://github.com/rafaelmardojai/firefox-gnome-theme)
        ''
        false;
```

I am unsure whether this is the idiomatic syntax for embedding URLs and whether multiline strings cause problems. Consider looking at how Home Manager expresses external references.

---

Nix seems to use Markdown for descriptions nowadays, so this should be fine. Some Home Manager modules also use this masked link syntax.